src/meta/server_state.h

```diff
@@ -292,6 +292,9 @@ class server_state
     void process_one_partition(std::shared_ptr<app_state> &app);
     void transition_staging_state(std::shared_ptr<app_state> &app);
 
+    // check whether a max replica count is valid especially for a new app
+    bool validate_target_max_replica_count(int32_t max_replica_count);
```

Comment:
private is OK?

Reply:
This means `validate_target_max_replica_count` should be defined as a private method ? Actually it is now

Reply:
OK, I missed that.